### PR TITLE
Improve lifecycle of Integration tests

### PR DESCRIPTION
**Pull request by see-quick**
This PR improves our Integration tests lifecycle. It re-implements the approach to **not USE** shared objects during the whole test execution. Therefore, we decided to create new instances of Kafka cluster, Bridge, etc., for each test suite. For edge cases where a different configuration is needed, I have added overridable hooks.

### CODE REVIEW

File: src/test/java/io/strimzi/kafka/bridge/http/base/HttpBridgeITAbstract.java [hunk 1]
```diff
@@ -183,4 +159,102 @@ protected String generateRandomConsumerGroupName() {
...
+     * &#064;Override
+     * protected Map&lt;String, Object&gt; overridableConfig() {
+     *     Map&lt;String, Object&gt; overrides = new HashMap&lt;&gt;();
+     *     overrides.put("bridge.id", "my-custom-bridge");
+     *     return overrides;
+     * }
+     * Return a map with configuration keys to override default values,
+     * or keys mapped to {@code null} to remove them.
+     */
+    protected Map<String, Object> overridableConfig() {
```
Comment (ppatierno) [inline src/test/java/io/strimzi/kafka/bridge/http/base/HttpBridgeITAbstract.java]: why not just `overrideConfig`?

**Review by ppatierno — APPROVED** [review]
I tried it locally and works fine (with podman). I left a nit on a method name.

File: src/test/java/io/strimzi/kafka/bridge/http/base/HttpBridgeITAbstract.java (hunk 1 above)
Comment (see-quick) [reply src/test/java/io/strimzi/kafka/bridge/http/base/HttpBridgeITAbstract.java]: Yeah, that's better. I will update it.
